### DISCUSSION TITLE
Seq 2023.1 consumed all the RAM without admitting it

### DISCUSSION THREAD

**Discussion by korggy** [datalust/seq-tickets#1760] [Help] [answered]
Hello,

We have been running previous versions of Seq for many years without issue. Last night we installed 2023.1 on our 39 GB VM running Server 2016 on a host machine running Server 2016. Everything was fine all night long, with Seq reporting (via Task Manager) RAM usage < 500 MB. This morning, overall RAM utilization suddenly jumped to 99% and applications stopped responding. We checked Task Manager but no process was using more than 500 MB, so we were initially confused. Seq was reporting RAM usage of only ~225 MB. On a hunch that the recent Seq upgrade had something to do with it, we restarted Seq and overall RAM usage dropped from 99% (39 GB) to ~20% (8 GB), where it remains as of now. So, while at no point did Seq report using more than 500 MB, somehow it actually was, because restarting the process freed up all the memory immediately.

Screenshots taking during the issue:
![image](https://user-images.githubusercontent.com/1387687/217557155-026fc5b9-231b-4c90-8205-e3f53a46afd7.png)
![image](https://user-images.githubusercontent.com/1387687/217557297-e1fe85cc-5069-4044-95a0-0c4db69d14d5.png)

Here is the memory usage while restarting the Seq process:
![image](https://user-images.githubusercontent.com/1387687/217565478-f0fa6d83-30bb-478a-b248-e3992f5d1c9c.png)

Any ideas for us to try? This is now happening roughly every hour.

Thank you,
Mike

**Comment by KodrAus** [comment, marked as answer]
Hi @korggy :wave:

Seq 2023.1 uses an entirely new storage architecture compared to 2022.1. Instead of filling RAM with its own in-memory event cache, it relies on the OS to manage memory-mapped files. This memory is in use, but isn't reported as working set for Seq because it can be freely reclaimed by the Seq or any other process that needs memory.

It sounds like this approach may not be playing nicely with other applications running on the same VM as Seq if you're seeing them stop responding. Have you noticed any instability in Seq itself while memory is peaked at 99%?

**Reply by tylerbabury** [reply]
Our server instances (2023.1.8829) are also affected by this issue. After maxing out resource consumption the application ceases to ingest more logs and the web portal no longer loads.

**Reply by nblumhardt** [reply]
Thanks for the note @tylerbabury. Upgrading to 2023.1.8899 may address the issue for you (a number of fixes in that patch), but if not, we're tracking a similar problem in https://github.com/datalust/seq-tickets/issues/1767 and expect to have a patch for it soon.

**Comment by nblumhardt** [comment]
See also: https://github.com/datalust/seq-tickets/issues/1762 --- we're aiming to add a setting for this in the next maintenance release.
